reactive-banana/src/Control/Monad/Trans/ReaderWriterIO.hs

```diff
@@ -11,87 +11,38 @@ module Control.Monad.Trans.ReaderWriterIO (
 import Control.Monad.Fix
 import Control.Monad.IO.Class
 import Control.Monad.Trans.Class
-import Data.IORef
+import Control.Monad.Trans.Writer.CPS ( WriterT )
+import qualified Control.Monad.Trans.Writer.CPS as W
+import Control.Monad.Trans.Reader ( ReaderT )
+import qualified Control.Monad.Trans.Reader as R
+import Data.Monoid
 
-{-----------------------------------------------------------------------------
-    Type and class instances
-------------------------------------------------------------------------------}
-newtype ReaderWriterIOT r w m a = ReaderWriterIOT { run :: r -> IORef w -> m a }
```

Comment:
This actually just incurs more boxing than using CPS'ed `WriterT`, because whenever we call `run` we need to box up an `IORef`